Origin: WMF
Issue: https://phabricator.wikimedia.org/T145661
Description: Do not split -sfile in 2^30 chunks. Used on varnish-be, it could be potentially dropped.
diff --git a/bin/varnishd/storage/storage_file.c b/bin/varnishd/storage/storage_file.c
index 143ab9d..58657a9 100644
--- a/bin/varnishd/storage/storage_file.c
+++ b/bin/varnishd/storage/storage_file.c
@@ -37,6 +37,7 @@
 
 #include <errno.h>
 #include <stdio.h>
+#include <stdint.h>
 #include <stdlib.h>
 
 #include "storage/storage.h"
@@ -402,7 +403,7 @@ static void __match_proto__(storage_open_f)
 smf_open(struct stevedore *st)
 {
 	struct smf_sc *sc;
-	off_t fail = 1 << 30;	/* XXX: where is OFF_T_MAX ? */
+	off_t fail = sizeof(off_t) == sizeof(int32_t) ? INT32_MAX : INT64_MAX;
 	off_t sum = 0;
 
 	ASSERT_CLI();
